Doc/whatsnew/3.12.rst

```diff
@@ -198,10 +206,25 @@ Pending Removal in Python 3.14
 
   (Contributed by Jason R. Coombs and Hugo van Kemenade in :gh:`93963`.)
 
+APIs:
+
+*  :meth:`asyncio.Future.cancel`'s *msg* parameter (:gh:`90985`)
+*  :meth:`asyncio.Task.cancel`'s *msg* parameter (:gh:`90985`)
+
+Pending Removal in Python 4.0
+-----------------------------
+
+The following have been deprecated in earlier Python releases,
+and will be removed in Python 4.0.
```

Comment:
Do we need to clarify what we mean with Python 4?

This seems to imply that
1. there will be a Python 4.0;
2. this API will be removed in Python 4.0

However my understanding is that "Python 4" means "we are not going to remove it any time soon, and if we are going to remove it all it will be in the distant future if/when Python 4 is released".

Reply:
Yup, we shouldn't document something about Python 4 when that's hypothetical and devoid of specific meaning.

If we want to raise DeprecationWarning before we document a 3.xy deprecation timeline (as suggested in https://github.com/python/cpython/pull/93986#discussion_r902734305 ), I'd vote we not document this deprecation until then.

Reply:
The deprecation itself should be documented, and I think it would be better to also document that the removal is not currently planned (exact wording TBD).

Reply:
In https://github.com/python/cpython/issues/80480, the removal was suggested for "4.0 or 3.10", when we didn't know which was coming next (https://github.com/python/cpython/issues/80480#issuecomment-1093818166):

> The doc says:
> 
> 'u' will be removed together with the rest of the Py_UNICODE API.
Deprecated since version 3.3, will be removed in version 4.0.
[docs.python.org/3/library/array.html](https://docs.python.org/3/library/array.html)
> 
> But DeprecationWarning is not raised yet. Let's raise it.
> 
> * 3.8 -- PendingDeprecationWarning
> * 3.9 -- DeprecationWarning
> * 4.0 or 3.10 -- Remove it.

And https://github.com/python/cpython/issues/80480#issuecomment-1093818167:

> '4.0' is a stand-in for 'sometime after 2.7.final', scheduled for Jan 2020. A Pending... for 3.8.0, scheduled for Oct 2019, seems reasonable to me. Perhaps we should have a pydev discussion for the general issue of post 2.7 removals of already deprecated items.

Please could one of you bump that issue with a suggestion on how to proceed? Maybe it can be removed in 3.12?

---

This PR is documenting the status quo, repeating the 4.0 removal already documented at https://docs.python.org/3/library/array.html to bring more visibility. (This discussion shows it's working! :)

> The deprecation itself should be documented, and I think it would be better to also document that the removal is not currently planned (exact wording TBD).

Agreed. For now, shall we move this to either of these sections?

* the "Pending Removal in Future Versions" section: "The following were deprecated in earlier Python versions and will be removed, although there is currently no date scheduled for their removal."

* or the top-level "Deprecated" section, with no intro text. There's `typing` deprecations here, which may never be removed.

And do we want to delete the 4.0 removal at https://docs.python.org/3/library/array.html?

Reply:
> Please could one of you bump that issue with a suggestion on how to proceed? Maybe it can be removed in 3.12?

I left a comment in #80480.  It seems the code doesn't raise a deprecation warning, so it will have to go through a deprecation cycle before it can be removed.  Once the warning is added, we will know in which release it will be removed and we will be able to update the doc accordingly.

> the "Pending Removal in Future Versions" section:

In the meanwhile I think it can be moved here.

Reply:
Moved!